platforms/windows/hax_entry.c

```diff
@@ -431,6 +431,18 @@ NTSTATUS HaxVcpuControl(PDEVICE_OBJECT DeviceObject,
             vcpu_debug(cvcpu, (struct hax_debug_t*)inBuf);
             break;
         }
+        case HAX_VCPU_IOCTL_SET_CPUID: {
+            struct hax_cpuid *cpuid = (struct hax_cpuid *)inBuf;
+            if (inBufLength == 0 || inBufLength < sizeof(struct hax_cpuid) +
+                cpuid->total * sizeof(struct hax_cpuid_entry)) {
```

Comment:
This is better:
if (inBufLength < sizeof(struct hax_cpuid) || inBufLength < sizeof(struct hax_cpuid) +
                cpuid->total * sizeof(struct hax_cpuid_entry)) {
total is first member with 4 bytes size. So any inBufLength less than 4 bytes will crash.

Reply:
Thanks for your point. Actually, below expressions should be more exact as sizeof(struct hax_cpuid) is 8. Anyway, I will adopt your suggestion.

`if (inBufLength < sizeof(cpuid->total) || ...)` or `if (inBufLength < sizeof(uint32_t) || ...)`